lib/HLSL/HLOperationLower.cpp

```diff
@@ -604,18 +604,15 @@ Value *TranslateD3DColorToUByte4(CallInst *CI, IntrinsicOp IOP,
   Value *val = CI->getArgOperand(HLOperandIndex::kUnaryOpSrc0Idx);
   Type *Ty = val->getType();
 
-  bool isFXCCompatMode = CI->getModule()->GetHLModule().GetHLOptions().bFXCCompatMode;
-  // Use literal 255.001953 for scaling when FXC compat mode is enabled as
-  // FXC uses this value instead of 255 for some uknown reason.
-  static const float scalingVal = isFXCCompatMode ? 255.001953 : 255.0;
-  Constant *toByteConst = ConstantFP::get(Ty->getScalarType(), scalingVal);
+  // Use the same scaling factor used by FXC (i.e., 255.001953)
```

Comment:
I'd put the rationale from stacko, maybe this quote is the clearest: `Built-in rounding, necessary because of truncation. 0.001953 * 256 = 0.5`

Reply:
Added. Also updated PR description.